Assets/Mirror/Runtime/NetworkConnectionToClient.cs

```diff
@@ -56,6 +63,31 @@ public override void Disconnect()
             isReady = false;
             Transport.activeTransport.ServerDisconnect(connectionId);
             RemoveObservers();
+            DestroyOwnedObjects();
+        }
+
+        internal void AddOwnedObject(NetworkIdentity obj)
+        {
+            clientOwnedObjects.Add(obj);
+        }
+
+        internal void RemoveOwnedObject(NetworkIdentity obj)
+        {
+            clientOwnedObjects.Remove(obj);
+        }
+
+        protected void DestroyOwnedObjects()
+        {
+            // work on copy because the list may be modified when we destroy the objects
+            HashSet<NetworkIdentity> objects = new HashSet<NetworkIdentity>(clientOwnedObjects);
```

Comment:
why is this copied?

Reply:
As I tested it without the copy,  it was finding a collection modified exception here.   It looks like the list is being updated while we are destroying the objects.